mslicedevel.bat

```diff
@@ -1,9 +1,59 @@
 ::
 :: Basic wrapper script for running in development mode.
-:: It assumes that mantid is installed in the standard
-:: location.
 ::
+@setlocal
 @set THIS_DIR=%~dp0
-@set PYTHON_EXE=C:\MantidInstall\bin\python.exe
 
-%THIS_DIR%\scripts\mslice.bat
+:: Define here the path to your conda installation and uncomment the line
+::@set CONDAPATH=
+
+:: ActiveEnv is set to true once the conda environment mantidnightly was successfully activated
+@set ActiveEnv=false
+
+:: With CONDAPATH set mantidnightly can be activated even when the base environment is not activated yet, for instance by doubleclicking on mslicedevel.bat
+@if defined CONDAPATH (
+    goto CheckBaseEnv
+) else if defined CONDA_DEFAULT_ENV (
+    :: Without CONDAPATH set it is still possible to activate mantidnightly when mslicedevel.bat is run from a conda prompt
+    goto ActivateEnv
+) else (
+    goto NoConda
+)
+
+:CheckBaseEnv
+@if defined CONDA_DEFAULT_ENV (
+    goto ActivateEnv
+) else (
+    :: Activate the base environment first if necessary
+    call %CONDAPATH%\Scripts\activate.bat %CONDAPATH%
+    goto ActivateEnv
+)
+
+:ActivateEnv
+:: Activate mantidnightly
+@if %CONDA_DEFAULT_ENV% == base (
+    call conda activate mantidnightly
+)
+@if %CONDA_DEFAULT_ENV% == mantidnightly (
+    @set ActiveEnv=true
+) else (
+    goto NoConda
+)
+
+:NoConda
+:: This allows to start MSlice when there is no conda available or activating mantidnightly failed
+@set MANTIDPATH=C:\MantidInstall\bin
```

Comment:
@MialLewis It is likely that a developer has quite a few versions of MantidInstall available. In this case setting the MantidInstall path manually might be the best option to ensure that the most suitable Mantid installation is used and not a completely outdated one.

Reply:
If someone creates their conda environment for this with `conda env create -f mslice-developer.yml` as described in `README` the conda environment will be called `mantidnightly` though. Modifying `mslicedevel.bat` or `mslicedevel.sh` manually is probably the easiest solution in case the conda environment has a different name.

Reply:
I don't think I was suggesting to look for the install location/conda env automatically here, I was just suggesting there was an obvious place to input them (top of the file?) and an error message output suggesting to do this if they are not found initially as the default values aren't used.

In file attached in the following comment I implement this, but for mamba path (have a obvious place at the top to input the file paths, helpful error message if this is required). The difference is there I search for the mamba directory via the path, as I think that's pretty safe to do.

Reply:
I see I did actually suggest this below, and agree looking for the envs and install is a bad idea